tests/test_yaml.py

```diff
@@ -0,0 +1,115 @@
+# This file is part of daf_base
+#
+# Developed for the LSST Data Management System.
+# This product includes software developed by the LSST Project
+# (http://www.lsst.org/).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+"""Test YAML serialization"""
+
+import os
+import unittest
+
+try:
+    import yaml
+except ImportError:
+    yaml = None
+
+import lsst.daf.base
+
+TESTDIR = os.path.abspath(os.path.dirname(__file__))
+
+
+class YAMLTestCase(unittest.TestCase):
+
+    @unittest.skipIf(yaml is None, "yaml module not installed")
+    def setUp(self):
+        pass
+
+    def assertEqualPL(self, pl1, pl2):
+        for name in pl1.getOrderedNames():
+            self.assertEqual(pl1.getArray(name), pl2.getArray(name))
+            self.assertEqual(pl1.getScalar(name), pl2.getScalar(name))
+            self.assertEqual(pl1.typeOf(name), pl2.typeOf(name))
+
+    def assertEqualPS(self, ps1, ps2):
+        self.assertEqual(ps1.nameCount(), ps2.nameCount())
+        self.assertEqual(set(ps1.paramNames(False)), set(ps2.paramNames(False)))
+        for name in ps1.paramNames(False):
+            self.assertEqual(ps1.getArray(name), ps2.getArray(name))
+            self.assertEqual(ps1.getScalar(name), ps2.getScalar(name))
+            self.assertEqual(ps1.typeOf(name), ps2.typeOf(name))
+
+    def testYamlPS(self):
+        ps = lsst.daf.base.PropertySet()
+        ps.setBool("bool", True)
+        ps.setShort("short", 42)
+        ps.setInt("int", 2008)
+        ps.setLongLong("int64_t", 0xfeeddeadbeef)
+        ps.setFloat("float", 3.14159)
+        ps.setDouble("double", 2.718281828459045)
+        ps.set("char*", "foo")
+        ps.setString("string", "bar")
+        ps.set("char*", u"foo")
+        ps.setString("string", u"bar")
+        ps.set("int2", 2009)
+        ps.set("dt", lsst.daf.base.DateTime("20090402T072639.314159265Z", lsst.daf.base.DateTime.UTC))
+        ps.set("blank", "")
```

Comment:
Please add both an entry that is a `PropertySet` and another that is a `PropertyList` and make sure both are retrieved using the correct type.

This is less important for testing `PropertyList` below, because `PropertyList` is flat, with dotted names. (Even so, it would not hurt to do that below as well).